src/main/java/com/amazon/opendistroforelasticsearch/security/resolver/IndexResolverReplacer.java

```diff
@@ -692,20 +692,7 @@ private boolean getOrReplaceAllIndices(final Object request, final IndicesProvid
             ((UpdateRequest) request).index(newIndices.length!=1?null:newIndices[0]);
         } else if (request instanceof SingleShardRequest) {
             final SingleShardRequest<?> gr = (SingleShardRequest<?>) request;
-            final String[] indices = gr.indices();
-            final String index = gr.index();
-
-            final List<String> indicesL = new ArrayList<String>();
-
-            if (index != null) {
-                indicesL.add(index);
-            }
-
-            if (indices != null && indices.length > 0) {
-                indicesL.addAll(Arrays.asList(indices));
-            }
-
-            String[] newIndices = provider.provide(indicesL.toArray(new String[0]), request, true);
+            String[] newIndices = provider.provide(gr.indices(), request, true);
```

Comment:
In case of `null` index, passing array with single element that is `null` does not sound right to me. Can't you pass `null` instead?

Reply:
I don't see a bug here as the code does the same thing in an undesirable way hence chose it to be refactoring. 
Single element `null` is handled by the inner code. There may be other requests suffering from same issue but will track it in separate issue.

Reply:
In case index is not `null`, how many elements will be in the `indicesL`?

Reply:
Passing `null` instead of array with single element `null` may not have the same [behavior](https://github.com/elastic/elasticsearch/blob/master/server/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java#L656). Hence don't want to change it.
If taking advantage of knowledge of underlying implementation, `indicesL` will be of size `1`.

Reply:
I don't understand reference to the [behavior](https://github.com/elastic/elasticsearch/blob/master/server/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java#L656). The function is called from `isAllWithNoRemote()` and `isLocalAll()` (did I miss other places?). Do you mean the the result of those 2 methods will be different if we pass `null` or an array with single `null` element? If not, what is the other side effect of passing null?

Reply:
This is with reference to your suggestion in #886 removing the null check. 
I don't know the side effects of passing null or changing behavior hence uncomfortable making the change.

Reply:
It goes in circle. In #886 the check for an array with single `null` element can't be removed as it is created and passed here.

I am -0 on the PR. I don't block it, as it is an improvement compared to the original code, but also don't approve it as I would prefer to see why `null` can't be passed instead of an array with single `null` element.